classes/db/DbPDO.php

```diff
@@ -127,6 +127,11 @@ public function connect()
             throw new PrestaShopException('Link to database cannot be established: ' . $e->getMessage());
         }
 
+        // UTF-8 support
+        if (!$this->link->query('SET NAMES \'utf8\'')) {
```

Comment:
We already have a `Db::checkEncoding` function to do that kind of stuff.
@mickaelandrieu it looks like yes, this is mandatory: https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Install/Database.php#L73

Reply:
@PierreRambaud it seems mandatory only during install. Many developers install prestashop locally and then import it to online server. Problems occur when encoding configuration is different there. Wouldn't it be better to check encoding also on every session and inform if something needs to be changed in current server configuration?

Reply:
then I guess this also could use checkEncoding function instead
https://github.com/PrestaShop/PrestaShop/blob/4b28feafd38cc52b210b9f55ea7ab0bc9cad4816/classes/db/DbMySQLi.php#L76-L79

Reply:
Yes of course, maybe it is time to do a refactoring and improve a little bit the code :smile:

Reply:
Hi, @SharakPL and @PierreRambaud , I  am glad that I am not the only one who have the same problem.

Calling checkEncoding (or tryUTF8) doesn't help because checkEncoding creates a new instance of PDO, then checks utf8 and then immediately closes the connection. This is not what we need. We need SET NAMES utf8 for every connection we establish and then work with. I have PR #11648 that fixes the same issue for me. I think Prestashop should accept PR of @SharakPL if it doesn't impact performance a lot and do refactoring afterwards if it is needed. Now PrestaShop requires UTF8 encoding which makes a strong limitation for hostings especially in non-English speaking countries. Usually, hostings are not able to change MySQL setting for just one user.

I am developing a small shop and I am forced to make my own patches into PrestaShop core. It's a big disadvantage of Prestashop and now we can fix it and remove utf8 requirement in the future.

Reply:
I know calling checkEncoding creates a new instance, that is exactly I asked for a refactoring / improvement  ^^
If you don't want to be impacted, we are also able to change your MySQL configuration to force utf8 by default, even if it isn't specify in your PDO configuration.